hazelcast/src/main/java/com/hazelcast/mapreduce/impl/task/DefaultContext.java

```diff
@@ -102,11 +102,12 @@ public void finalizeCombiners() {
         Combiner<ValueIn, ?> combiner = combiners.get(key);
         if (combiner == null) {
             combiner = combinerFactory.newCombiner(key);
```

Comment:
It is a good fix - but it would be more efficient to omit combinerFactory.newCombiner(key); it case if key exists.
We can add computeIfAbsent into out HazelCast ConcurrentHashMap SampleableConcurrentHashMap.
And use it in a lot of similar places. It would reduce unnecessary objects creation.